intera_interface/src/intera_interface/limb.py

```diff
@@ -519,3 +588,147 @@ def test_collision():
+        @type nullspace_gain: double
+        @param nullspace_gain: gain used to bias toward the nullspace goal [0.0, 1.0] (optional)
+        @rtype: dict({str:float})
+        @return: valid joint positions if exists.  False if no solution is found.
+        """
+        # Add desired pose for inverse kinematics
+        ikreq = SolvePositionIKRequest()
+        hdr = Header(stamp=rospy.Time.now(), frame_id='base')
+        ikreq.pose_stamp.append(PoseStamped(header=hdr, pose=pose))
+        ikreq.tip_names.append(end_point)
+
+        # The joint seed is where the IK position solver starts its optimization
+        if joint_seed is not None:
+            ikreq.seed_mode = ikreq.SEED_USER
+            seed = JointState()
+            seed.name = self._joint_names
+            seed.position = joint_seed
+            ikreq.seed_angles.append(seed)
+
+        # Once the primary IK task is solved, the solver will then try to bias the
+        # the joint angles toward the goal joint configuration. The null space is
+        # the extra degrees of freedom the joints can move without affecting the
+        # primary IK task.
+        if nullspace_goal is not None:
+          ikreq.use_nullspace_goal.append(True)
+          # The nullspace goal can either be the full set or subset of joint angles
+          goal = JointState()
+          goal.position = joint_seed
+          if nullspace_joint_names is None:
+              goal.name = self._joint_names
+          else:
+              goal.name = nullspace_joint_names
+          ikreq.nullspace_goal.append(goal)
+          # The gain used to bias toward the nullspace goal. Must be [0.0, 1.0]
+          # If empty, the default gain of 0.4 will be used
+          ikreq.nullspace_gain.append(nullspace_gain)
+
+        try:
+            resp = self._iksvc(ikreq)
+        except (rospy.ServiceException, rospy.ROSException), e:
+            rospy.logerr("IK Service call failed: %s" % (e,))
+            return False
+        limb_joints = {}
+        # Check if result valid, and type of seed ultimately used to get solution
+        if (resp.result_type[0] > 0):
+            seed_str = {
+                        ikreq.SEED_USER: 'User Provided Seed',
+                        ikreq.SEED_CURRENT: 'Current Joint Angles',
+                        ikreq.SEED_NS_MAP: 'Nullspace Setpoints',
+                       }.get(resp.result_type[0], 'None')
+            # Format solution into Limb API-compatible dictionary
+            limb_joints = dict(zip(resp.joints[0].name, resp.joints[0].position))
+        else:
+            rospy.logerr("INVALID POSE - No Valid Joint Solution Found.")
+            return False
+        return limb_joints
+
+    def fk_request(self, joint_angles,
+                        end_point='right_hand'):
+        """
+        Forward Kinematics request sent to FK Service
+
+        @type joint_angles: [double]
+        @param joint_angles: the arm's joint positions
```

Comment:
It's currently ordered in the joint order base to tip (j0 to j6). There's already a lot of packing and unpacking dicts which is probably not needed if we just decide on a consistent ordering.